app/frontend/src/features/profile/edit/EditProfile.tsx

```diff
@@ -124,17 +130,35 @@ export default function EditProfileForm() {
     null
   );
   const queryClient = useQueryClient();
-  const { control, errors, register, handleSubmit, setValue } =
+  const { control, errors, register, handleSubmit, setValue, formState } =
     useForm<FormValues>({
       defaultValues: {
         city: user?.city,
         lat: user?.lat,
         lng: user?.lng,
         radius: user?.radius,
+        name: user?.name,
+        hometown: user?.hometown,
+        pronouns: user?.pronouns,
+        occupation: user?.occupation,
+        education: user?.education,
+        aboutMe: user?.aboutMe,
+        myTravels: user?.myTravels,
+        thingsILike: user?.thingsILike,
+        hostingStatus: user?.hostingStatus,
+        meetupStatus: user?.meetupStatus,
+        fluentLanguages: user?.languageAbilities,
+        regionsVisited: user?.regionsVisited,
+        regionsLived: user?.regionsLived,
+        additionalInformation: user?.additionalInformation,
+        avatarKey: user?.avatarKey,
       },
       shouldFocusError: true,
     });
 
+  const { isDirty } = formState;
+  setFormIsDirty(isDirty);
+
   //Although the default value was set above, if the page is just loaded,
   //user will be undefined on first render, so the default values will be undefined.
   //So make sure to set values when user finshes loading
```

Comment:
This is the problem - this `useEffect` below always calls `setValue` and the form is then dirty.

The problem is these below useEffects are workarounds for the fact that `city`, `lat`, `lng`, and `radius` are not fields in the form, but are set by the `location` field.

There's a much better way to do it: firstly, delete the below `useEffects`, then change the location controller to something like this:
```
          <Controller
            defaultValue={{
              address: user.city,
              lat: user.lat,
              lng: user.lng,
              radius: user.radius,
            }}
            name="location"
            control={control}
            render={(onChange) => (
              <EditLocationMap
                showRadiusSlider
                initialLocation={{
                  address: user.city,
                  lat: user.lat,
                  lng: user.lng,
                  radius: user.radius,
                }}
                updateLocation={(location) => {
                  if (location) {
                    onChange({
                      city: location.address,
                      lat: location.lat,
                      lng: location.lng,
                      radius: location.radius,
                    });
                  }
                }}
              />
            )}
          />
```

Then change `handleSubmit` so all the fields are passed to `updateUserProfile`. Also get rid of all the `defaultValues` in `useForm` (remember, user might be `null` on first render anyway so it is currently setting all default values to null. When `defaultValue` is defined on the fields, user already exists since the fields aren't displayed until user is defined)

That's a bit complex, let me know on slack if you need further explanation.